src/graphemebreak.c

```diff
@@ -91,8 +91,37 @@ static enum GraphemeBreakClass get_char_gb_class(utf32_t ch)
     return GBP_Other;
 }
 
+/**
+ * find out if a codepoint is an extended pictographic point
+ *
+ * @param ch   character to check
+ * @return     true, if the codepoint is extended pictographic
+ */
+static bool is_char_extended_pictographic(utf32_t ch)
+{
+    int min = 0;
+    int max = ARRAY_LEN(ep_prop) - 1;
+    int mid;
+
+    do
+    {
+        mid = (min + max) / 2;
+
+        if (ch < ep_prop[mid].start)
+            max = mid - 1;
+        else if (ch > ep_prop[mid].end)
+            min = mid + 1;
+        else
+            return true;
+    } while (min <= max);
+
+    return false;
+}
+
 /**
  * Sets the grapheme breaking information for a generic input string.
+ * It uses the extended grapheme cluster ruleset, except for rule 9c. Rule 9c only
+ * get relevant when implementing CLDR, which we don't at the moment
```

Comment:
"gets", I think? Also two spaces after the full stop is preferred.

Reply:
Maybe we can delete these two lines. GB9c does not seem to make into the final TR29: http://www.unicode.org/reports/tr29/.